Hydration_freenrg/python/MinEq_solv.py

```diff
@@ -0,0 +1,165 @@
+import BioSimSpace as BSS
+import argparse
+import os
+
+"""Script to Minimise and Equilibrate solvated systems according to the nine step
+conservative protocol defined in https://doi.org/10.1063/5.0013849
+designed to be run in the respective replica folder
+"""
+#BSS.verbose(True)
+solvated = BSS.IO.readPerturbableSystem(
+    top0="pertsave_solv0.prm7",
+    coords0="pertsave_solv0.rst7",
+    top1="pertsave_solv1.prm7",
+    coords1= "pertsave_solv1.rst7",
+)
+#Need to manually change the device index in each CUDA equilibration to 0
+#stops wierd interactions with SLURM 
+#(can be done with CUDA_VISIBLE_DEVICES instead but this is easier for large scripts)
+substring = "CudaDeviceIndex"
+
+minimise_1 = BSS.Protocol.Minimisation(
+    steps=1000, restraint="heavy", force_constant=5.0
+)
+process_m1 = BSS.Process.OpenMM(solvated, minimise_1, work_dir="Minimise_1")
+process_m1.start()
+process_m1.wait()
+min1 = process_m1.getSystem()
```

Comment:
Make sure to check that the process didn't raise an error and that the system returned is not `None`, e.g.:

```python
if process_m1.isError()
    raise SomeException("...")

min1 = process_m1.getSystem()

if min1 is None:
    raise SomeException("...")
```

(The system could be `None` due to the particular protocol setup.)

These checks should be added in other scripts too, but I won't duplicate.

Reply:
I've added `MinimisationError,` `EquilibrationError` and `GetSystemError` exceptions to all minimisation and equilibration scripts, checked after each process.

Reply:
Great, that sounds good!